### PR TITLE
[docs] Move 2024 migration guide

### PR DESCRIPTION
**Pull request by tnowacki**
## Description 

- One pager migration guide 

## Test Plan 

- 👀 

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: docs/content/guides/developer/advanced/move-2024-migration.mdx [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+### public(package) {#public-package}
+
+`friend` declarations, and the associated `public(friend)` visibility modifiers, are deprecated. In their place is the `public(package)` visibility modifier, which allows calling functions only within the same package where they are defined.
+
+### Positional fields
+
+You can now define `struct`s with positional fields, which are accessed by zero-based index. For example,
+
+```move
+public struct Pair(u64, u64) has copy, drop, store.
```
Comment (ronny-mysten) [inline docs/content/guides/developer/advanced/move-2024-migration.mdx:77]: I believe that period shouldn't be there?

File: docs/content/guides/developer/advanced/move-2024-migration.mdx [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+                             // ^ note `::` here
+```
+
+The migration script attempts to remediate naming errors using global qualification when possible.
+
+### Follow along
+
+The beta release of Move 2024 comes with some powerful new features beyond the breaking changes described here. There are also more on the horizon. Join the Sui newsletter to learn about new, exciting features coming to Move this year, including syntactic macros, enums with pattern matching, and other user-defined syntax extensions.
+
+Please, provide any feedback or report any issues you encounter via GitHub, Discord, or Telegram.
```
Comment (ronny-mysten) [inline docs/content/guides/developer/advanced/move-2024-migration.mdx:300]: Not sure the best place to direct these.

File: docs/content/guides/developer/advanced/move-2024-migration.mdx [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+...
+#[expected_failure(abort_code = ::sui_system::validator_set::EInvalidCap)]
+                             // ^ note `::` here
+```
+
+The migration script attempts to remediate naming errors using global qualification when possible.
+
+### Follow along
+
+The beta release of Move 2024 comes with some powerful new features beyond the breaking changes described here. There are also more on the horizon. Join the Sui newsletter to learn about new, exciting features coming to Move this year, including syntactic macros, enums with pattern matching, and other user-defined syntax extensions.
```
Comment (ronny-mysten) [inline docs/content/guides/developer/advanced/move-2024-migration.mdx:298]: Note to self...get link for newsletter.

File: docs/content/guides/developer/advanced/move-2024-migration.mdx [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+
+This document highlights some new features to try out and shows how to migrate your existing modules to use Move 2024.
+
+## How to migrate
+
+To migrate a project to Move 2024 Beta, you have two options:
+
+1. Delete your existing `Move.lock` file (if one exists) to make sure you're using the newest `sui-framework` version.
+2. Run `sui move migrate` in the root of your Move project. See [Automatic migration](#automatic-migration). 
+   1. Alternatively, update your `Move.toml` file's `[package]` entry to include `edition = "2024.beta"`. If you do this, you might receive a number of new errors as part of our [breaking changes](#breaking-changes).
```
Comment (ronny-mysten) [inline docs/content/guides/developer/advanced/move-2024-migration.mdx:20]: This seems like three options. Checking offline...

File: docs/content/guides/developer/advanced/move-2024-migration.mdx (hunk 1 above)
Comment (tnowacki) [inline docs/content/guides/developer/advanced/move-2024-migration.mdx:77]:
```suggestion
public struct Pair(u64, u64) has copy, drop, store;
```

Comment (tnowacki) [reply docs/content/guides/developer/advanced/move-2024-migration.mdx:77]: Semi-colon, but good catch!

File: docs/content/guides/developer/advanced/move-2024-migration.mdx (hunk 2 above)
Comment (tnowacki) [reply docs/content/guides/developer/advanced/move-2024-migration.mdx:300]: Yeah I just listed these out arbitrarily. I'll defer to others for the feedback links here (wherever you want to send 'em, let's send 'em there)

**Review by ronny-mysten — APPROVED** [review]

**Pull request merged**